### PR TITLE
DIV-4945 Adding solicitor permissions for PETSOL role

### PR DESCRIPTION
**Pull request by alexwakeman**
[Solicitor can only see events in CCD relative to whom they represent](https://tools.hmcts.net/jira/browse/DIV-4945)

Also allows Linking for Respondent Solicitors from AosAwaiting (when Petitioner / + Solicitor has nominated the Resp. Solicitor)

------
ALSO
[Add Respondent Solicitor Email field to CCD](https://tools.hmcts.net/jira/browse/DIV-5610)

### CODE REVIEW

File: definitions/divorce/json/AuthorisationCaseEvent.json [hunk 1]
```diff
@@ -5157,5 +5157,26 @@
...
     "CRUD": "R"
+  },
+  {
+    "LiveFrom": "01/03/2019",
+    "CaseTypeID": "DIVORCE",
+    "CaseEventID": "solicitorApplyForDN",
+    "UserRole": "caseworker-divorce-solicitor",
+    "CRUD": "R"
+  },
+  {
```
Comment (alexwakeman) [inline definitions/divorce/json/AuthorisationCaseEvent.json]: Add read-only permissions so any solicitor can see events in History tab

File: charts/div-ccd-definitions/values.preview.template.yaml [hunk 2]
```diff
@@ -58,6 +58,7 @@ ccd:
         - divorce_as_caseworker_superuser@mailinator.com|DIVORCE|DIVORCE|Submitted
         - divorce_as_caseworker_la@mailinator.com|DIVORCE|DIVORCE|Submitted
         - divorce_as_caseworker_admin@mailinator.com|DIVORCE|DIVORCE|Submitted
+        - divorce_respondent_solicitor@mailinator.com|DIVORCE|DIVORCE|Submitted
```
Comment (yarpo) [inline charts/div-ccd-definitions/values.preview.template.yaml]: what does it do?

File: definitions/divorce/json/AuthorisationCaseEvent.json [hunk 3]
```diff
@@ -4525,7 +4532,7 @@
     "LiveFrom": "01/03/2019",
     "CaseTypeID": "DIVORCE",
     "CaseEventID": "solicitorApplyForDN",
-    "UserRole": "caseworker-divorce-solicitor",
```
Comment (yarpo) [inline definitions/divorce/json/AuthorisationCaseEvent.json]: are we adding permissions to PETSOL, or we replace caseworker-divorce-solicitor with PETSOL?

Is this correct that all other caseworker-divorce-solicitor will not be able to do it anymore?

File: definitions/divorce/json/AuthorisationCaseEvent.json (hunk 1 above)
Comment (yarpo) [reply definitions/divorce/json/AuthorisationCaseEvent.json]: ok, nevermind my prev question :)

File: charts/div-ccd-definitions/values.preview.template.yaml (hunk 2 above)
Comment (alexwakeman) [reply charts/div-ccd-definitions/values.preview.template.yaml]: It adds an extra solicitor user - we need a respondent solicitor for testing

Comment (yarpo) [reply charts/div-ccd-definitions/values.preview.template.yaml]: ok

**Pull request merged**